File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/api/PublishingHouseApi.java
package ru.spring5.elena.librarymanager.api;

import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import ru.spring5.elena.librarymanager.api.dto.Edition;
import ru.spring5.elena.librarymanager.api.dto.PublishingHouse;

public interface PublishingHouseApi {

	PublishingHouse getPublishingHouse(
			@NotNull Long publishingHouseId
	);

	Page<PublishingHouse> searchPublishingHouses(
			Pageable pageable
	);

	PublishingHouse createPublishingHouse(
			@Valid PublishingHouse publishingHouse
	);

	PublishingHouse savePublishingHouse(
			@NotNull Long publishingHouseId, 
			@Valid PublishingHouse publishingHouse
	);

	Page<Edition> searchEditions(
			@NotNull Long publishingHouseId, 
			Pageable pageable
	);

	List<Edition> updateEditionsByPublishingHouseId(
			@NotNull Long publishingHouseId, 
			@NotNull List<Long> editionIds
	);

}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/PublishingHouseApi.kt
package ru.elenacherev.librarymanager.api

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import ru.elenacherev.librarymanager.api.dto.Edition
import ru.elenacherev.librarymanager.api.dto.PublishingHouse
import java.util.*
import javax.validation.Valid

interface PublishingHouseApi {

    fun getPublishingHouse(
        publishingHouseId: UUID
    ): PublishingHouse?

    fun searchPublishingHouses(
        pageable: Pageable
    ): Page<PublishingHouse>

    fun createPublishingHouse(
        publishingHouse: @Valid PublishingHouse
    ): PublishingHouse

    fun savePublishingHouse(
        publishingHouseId: UUID,
        publishingHouse: @Valid PublishingHouse
    ): PublishingHouse

    fun searchEditions(
        publishingHouseId: UUID,
        pageable: Pageable
    ): Page<Edition>

    fun updateEditionsByPublishingHouseId(
        publishingHouseId: UUID,
        editionIds: List<UUID>
    ): List<Edition>
}